plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/DeltaLakeSchemaSupport.java

```diff
@@ -276,9 +281,41 @@ private static ColumnMetadata mapColumn(TypeManager typeManager, JsonNode node)
                 .setName(fieldName)
                 .setType(buildType(typeManager, typeNode))
                 .setNullable(nullable)
+                .setComment(getComment(node))
                 .build();
     }
 
+    public static Map<String, Optional<String>> getColumnComments(MetadataEntry metadataEntry)
```

Comment:
Is there a difference between missing entry and entry being `Optional.empty()`?

maybe we just have a map of non-null comments, passed around  as `Map<String,String>` ?

we would lose ability to validate map contains entries for the columns, but you don't do this anyway (by means of `columnComments.getOrDefault...`)